cirq/ops/controlled_gate.py

```diff
@@ -25,30 +25,40 @@
 class ControlledGate(raw_types.Gate):
     """Augments existing gates with a control qubit."""
 
-    def __init__(self, sub_gate: raw_types.Gate) -> None:
+    def __init__(self, sub_gate: raw_types.Gate,
+                 control_qubit: Optional[raw_types.QubitId] = None) -> None:
```

Comment:
I think that, instead of the class radically changing function based on this argument, that we should have two parameters: `control_qubits` and `num_unspecified_control_qubits`. Then the class can do both things, and it will always be possible to automatically flatten nested controlled gates.

Reply:
Done. Don't know why pytest2 fails.

Reply:
Fixed